schema/query/projection_evaluator.go

```diff
@@ -0,0 +1,170 @@
+package query
+
+import (
+	"context"
+	"fmt"
+
+	"sync"
+
+	"github.com/rs/rest-layer/schema"
+)
+
+// Eval evaluate the projection on the given payload with the help of the
+// validator. The resolver is used to fetch payload of references outside of the
+// provided payload.
+func (p Projection) Eval(ctx context.Context, payload map[string]interface{}, validator schema.Validator, resolver ReferenceResolver) (map[string]interface{}, error) {
```

Comment:
Another idea:

```go
type Items interface {
    Payloads() []map[string]interface{}
}

type Resource interface {
    Find(ctx context.Context, query query.Query) (Items, error)
    MultiGet(ctx context.Context, ids []interface{}) (Items, error)
    GetSubResource(path string) (Resource, error)
    Validator() schema.Validator
}

func (p Projection) Eval(ctx context.Context, payload map[string]interface{}, rsc Resource) (map[string]interface{}, error) {}
```

This solves another issue `MultiGet`. I currently "hack" the find method to simulate a multi-get, but as commented in the code, this is dangerous because it assumes the `id` in the payloads can be compared using equality with the queried ids, which might not always be the case.

With those interface, it should be easy to make the current `resource.Resource` type compatible without intermediary type, which is better for performance.